File: dart_core/test/token_cache_test.dart
import 'package:telebirr_sdk_core/src/token_cache.dart';
import 'package:test/test.dart';

void main() {
  group('TokenCache', () {
    test('returns null when empty', () {
      final cache = TokenCache();
      expect(cache.get(), isNull);
    });

    test('returns cached token', () {
      final cache = TokenCache();
      cache.set('test-token');
      expect(cache.get(), 'test-token');
    });

    test('returns null after TTL expires', () async {
      final cache = TokenCache(ttl: Duration(milliseconds: 1));
      cache.set('test-token');
      await Future<void>.delayed(Duration(milliseconds: 10));
      expect(cache.get(), isNull);
    });

    test('clears token', () {
      final cache = TokenCache();
      cache.set('test-token');
      cache.clear();
      expect(cache.get(), isNull);
    });
  });
}

File: ts-core/tests/token-cache.test.ts
import { TokenCache } from '../src/token-cache.js';

describe('TokenCache', () => {
  it('returns null when empty', () => {
    const cache = new TokenCache();
    expect(cache.get()).toBeNull();
  });

  it('returns cached token', () => {
    const cache = new TokenCache();
    cache.set('test-token');
    expect(cache.get()).toBe('test-token');
  });

  it('returns null after TTL expires', () => {
    const cache = new TokenCache({ ttlMs: 1 });
    cache.set('test-token');
    // Wait for TTL to expire
    return new Promise((resolve) => setTimeout(resolve, 10)).then(() => {
      expect(cache.get()).toBeNull();
    });
  });

  it('clears token', () => {
    const cache = new TokenCache();
    cache.set('test-token');
    cache.clear();
    expect(cache.get()).toBeNull();
  });

  it('uses default TTL of 50 minutes', () => {
    const cache = new TokenCache();
    cache.set('test-token');
    // Should still be valid immediately
    expect(cache.get()).toBe('test-token');
  });
});
